File: BackEnd/Service_AnimalCare/Service/Models/BodyResponse.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Service.Models
{
    public abstract class BodyResponse
    {
        public object Data { get; set; }
        public bool IsError { get; set; }
        public List<string> Errors { get; set; }
        public bool IsTokenTimeout { get; set; }

        public BodyResponse()
        {
            Data = null;
            IsError = false;
            Errors = new List<string>();
            IsTokenTimeout = false;
        }
    }

    public class PhanCongResponse : BodyResponse { }
}

File: BackEnd/Service_ImportExport/src/java/Models/BodyResponse.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Models;

import com.google.gson.Gson;
import java.util.*;
import javax.ws.rs.core.Response;

/**
 *
 * @author Shin'sLaptop
 */
public abstract class BodyResponse {
    public Object Data;
    public boolean IsError;
    public ArrayList<String> Errors;
    public boolean IsTokenTimeout;
    
    public BodyResponse(){
        Data = null;
        IsError = false;
        Errors = new ArrayList<>();
        IsTokenTimeout = false;
    }
    public Response json(){
        return Response.ok(new Gson().toJson(this))
                .header("Access-Control-Allow-Origin", "*")
                .build();
    }
    
}